resources/aws_ec2.go

```diff
@@ -663,3 +714,365 @@ func (obj *AwsEc2Res) UnmarshalYAML(unmarshal func(interface{}) error) error {
 func (obj *AwsEc2Res) prependName() string {
 	return AwsPrefix + obj.GetName()
 }
+
+// snsServer returns an http server used to listen for sns messages
+func (obj *AwsEc2Res) snsServer() *http.Server {
+	port := fmt.Sprintf(":%d", int(obj.SnsPort))
+	srv := &http.Server{Addr: port}
+	log.Printf("%s: Starting SNS Endpoint", obj)
+	//rand.Seed(time.Now().UTC().UnixNano())
+	http.HandleFunc("/", obj.postHandler)
+	log.Printf("%s: Listening on %s", obj, port)
+	go func() {
+		if err := srv.ListenAndServe(); err != nil {
+			select {
+			case obj.awsChan <- &chanStruct{
+				err: errwrap.Wrapf(err, "error starting http server"),
+			}:
+			case <-obj.closeChan:
+			}
+			return
+		}
+	}()
+	return srv
+}
+
+// postHandler listens for posts on the SNS Endpoint
+// TODO: download pem and check message against signature
+func (obj *AwsEc2Res) postHandler(w http.ResponseWriter, req *http.Request) {
+	if req.Method == "POST" {
+		post, _ := ioutil.ReadAll(req.Body)
```

Comment:
apparently there is a cleaner way to read that streams instead. In case this is an easy change and it's better, can you look into it?